\thispagestyle{plain}
\begin{center}
    \textbf{Abstract}
\end{center}
Distributions of splicing likelihood scores for single nucleotide variants 
(SNVs) were analyzed with the hopes that these scores could be used to 
predict their pathogenicity, defined as a pathogenicity score multiplied by a 
risk score for that gene. SNVs implicated in autism as well as those found in
control phenotypes were gathered and scored for their likelihood to cause 
cassette exon skipping and their pathogenicity. Gene coexpression networks
were constructed for affected genes per eight developmental periods using
publicly available RNA-seq data. Genes were scored for risk, per period, 
based on relative numbers of coexpression partners. A poserior probability 
function was calculated as the predictive model relating splice score to
disease phenotype likelihood and was benchmarked for predicitve power.
